src/awkward/highlevel.py

```diff
@@ -181,18 +183,6 @@ def __init__(
             layout = data._layout
             behavior = ak._util.behavior_of(data, behavior=behavior)
 
-        elif numpy.is_own_array(data) and data.dtype != np.dtype("O"):
-            layout = ak.operations.from_numpy(data, highlevel=False)
-
-        elif ak._nplikes.Cupy.is_own_array(data):
-            layout = ak.operations.from_cupy(data, highlevel=False)
-
-        elif ak._nplikes.Jax.is_own_array(data):
-            layout = ak.operations.from_jax(data, highlevel=False)
-
-        elif ak._util.in_module(data, "pyarrow"):
-            layout = ak.operations.from_arrow(data, highlevel=False)
-
         elif isinstance(data, dict):
```

Comment:
Now this logic goes through `to_layout` after the addition of a new `regulararray` argument.